onnxscript/function_libs/torch_lib/ops/nn.py

```diff
@@ -2197,85 +2197,99 @@ def aten_unflatten_dense_tensors(
-def aten_upsample_bicubic2d(
-    self: TReal,
-    output_size: INT64,
-    align_corners: bool,
-    scale_factors: Optional[TFloat] = None,
-) -> TReal:
-    """upsample_bicubic2d.vec(Tensor input, SymInt[]? output_size, bool align_corners, float[]? scale_factors) -> Tensor
-    upsample_bicubic2d(Tensor self, SymInt[2] output_size, bool align_corners, float? scales_h=None, float? scales_w=None) -> Tensor
-    """
+def _get_upsample_align_corners_mode(align_corners: bool) -> str:
+    return "align_corners" if align_corners else "pytorch_half_pixel"
 
-    if output_size is not None:
-        result = _aten_upsample_output_size(self, output_size, align_corners, "cubic")
-    else:
-        result = _aten_upsample_scales(self, scale_factors, align_corners, "cubic")
-    return result
 
-
-@torch_op("aten::upsample_bicubic2d", private=True)
+@torch_op(("aten::upsample_bicubic2d", "aten::upsample_bilinear2d"), private=True)
 def _aten_upsample_output_size(
     self: TReal,
     output_size: INT64,
-    align_corners: bool,
-    str_mode: str,
+    mode: str,
+    coordinate_transformation_mode: str,
 ) -> TReal:
     self_shape = op.Shape(self)
     starts = op.Constant(value_ints=[0])
     ends = op.Constant(value_ints=[2])
     batch_channel = op.Slice(self_shape, starts, ends)
     output_size = op.Concat(batch_channel, output_size, axis=0)
-    if align_corners:
-        result = op.Resize(
-            self,
-            None,
-            None,
-            output_size,
-            mode=str_mode,
-            coordinate_transformation_mode="align_corners",
-        )
-    else:
-        result = op.Resize(
-            self,
-            None,
-            None,
-            output_size,
-            mode=str_mode,
-            coordinate_transformation_mode="pytorch_half_pixel",
-        )
-
-    return result
+    return op.Resize(
+        self,
+        None,
+        None,
+        output_size,
+        mode=mode,
+        coordinate_transformation_mode=coordinate_transformation_mode,
+        nearest_mode="floor",
+    )
 
 
-@torch_op("aten::upsample_bicubic2d", private=True)
+@torch_op(("aten::upsample_bicubic2d", "aten::upsample_bilinear2d"), private=True)
 def _aten_upsample_scales(
     self: TReal,
     scale_factors: TFloat,
-    align_corners: bool,
-    str_mode: str,
+    mode: str,
+    coordinate_transformation_mode: str,
 ) -> TReal:
     scale_factors = op.Cast(scale_factors, to=FLOAT.dtype)
     scale_factors = op.Concat(op.Constant(value_floats=[1.0, 1.0]), scale_factors, axis=0)
-    if align_corners:
-        result = op.Resize(
+    return op.Resize(
+        self,
+        None,
+        scale_factors,  # format should be: [1.0, 1.0, scale_h, scale_w]
+        None,
+        mode=mode,
+        coordinate_transformation_mode=coordinate_transformation_mode,
+        nearest_mode="floor",
+    )
+
+
+@torch_op("aten::upsample_bicubic2d", trace_only=True)
```

Comment:
Curious of how we found this overload in the begining? It's not in https://github.com/pytorch/pytorch/blob/78a6b0c4793d93d0a9105d9c92e7b88794016e66/aten/src/ATen/native/native_functions.yaml though.

Reply:
Is it this https://github.com/pytorch/pytorch/blob/78a6b0c4793d93d0a9105d9c92e7b88794016e66/aten/src/ATen/native/native_functions.yaml#L12617 ?